### DISCUSSION TITLE
Suppressed carrier AM (or DSB) TX

### DISCUSSION THREAD

**Discussion by sfotis** [egzumer/uv-k5-firmware-custom#260] [Ideas]
I just read a little the code, and it seems that the power level can be controlled by an 8-bit register. By modifying according to modulation this register when TX'ing under SSB modulation would this be able to produce a suppressed carrier DSB signal ? Could the CPU handle the workload ? Would the RF chip accept such rapid changes in power level (I guess we need maybe few khz for decent voice quality). I did some checks modifying some wav files to 8 bit unsigned PCM and the voice quality was not so bad (in order to check how would voice be sampled using 8 bit amplitude levels). What do you think ?

**Comment by egzumer** [comment]
In my opinion this is never going to work as it should. No matter what. This is an FM transciever, even AM or SSB demodulation is out of the specs, not documented features, and all of the sudden people want AM or SSB transmittion to work.
I didn't look for tools for that so I might be wrong, but as far as I know the CPU has no access to anything audio related. No mic access, no speaker access.

**Comment by sfotis** [comment]
Got my hands dirty and started to experiment. I modified the app main loop, and when in TX , I call BK4819_GetVoiceAmplitudeOut() which provides the current MIC AF Amplitude. I modulated the BK4819_SetupPowerAmplifier bias value using the above value normalised, and i seem to have a working variable amplitude signal, but on the current carrier frequency. I also muted the TX Audio on the TX preparation routine, in order to avoid FM modulation. In order to be 100% correct I would have to use fast fourier transform to get the frequency of AF, in order to modify the center frequency accordingly, but I think the hardware is not capable for this. I think it could work by enabling BK4819 Register 40 (see register documentation for the above) and try to deviate the carrier accordingly and create a DSB signal cheaply. Lets see where this takes us !

**Comment by Hunterfnet** [comment]
The experimenting is the future for all , this cheap radio is the example . I agree about your perseverance and i like read any improvement or not ,  on this aspect. 
I just starting comparing with a "not real" SSB on other firmware and for now i'm very perplex on this way because  the SSB here is not a real DSB mode on this radio because allways exist the fixed carrier wave on silence that betrays it. I don't understand if it's a pseudo DSB or other mixed audio manipulation . Good know . On others aspects is a very good idea , if one can do , compare transmissions daily on free time with diverse make-concepts and ideas for make every thing and idea better . Thank you

**Comment by sfotis** [comment]
You may find a test firmware with my initial results here :
https://github.com/sfotis/uv-k5-firmware-ssbtx/releases

Code exists in master branch.

**Reply by Hunterfnet** [reply]
Thank you for sharing, I really appreciate it sfotis. I don't expect more practic development from this site.

**Comment by Hunterfnet** [comment]
Now i'm more perplex after see this video making a contact with LSB -DSB - after on FM on a perfect way , on appreciate when changes to FM how sounds the modulation . I know too that have created the mode CW for transmission too . I'ts incredible with this little chiness. I have no words .I need one quansheng more now , i have the 2 on real use , one more for only test .
https://www.youtube.com/watch?v=6dIe1DwWlG4

**Comment by KodiUsA** [comment]
congratulations for your continuous idea with this firmware
this actual project seems stopped for weeks and no seen any evolution unless very useless lilltle "superfluous" modifications 
the still unresolved mistakes and bad improved content on main screen makes still looks like an unfinished toy , bad for approve a programming project not looking professional at all. 
i think the main concept that makes this firmware still actually online is the spectrum , if that was not the case , surely it will be forgiven like the others olders projects.
go forward with these new ideas DSB CW maybe a correct AM emulation and overall a professional screen , we dont care 100 freq. steeps or a FM icon if the main content still half performed and it's not logical . 
Sorry for sincerity guys but you have a better opponent that makes the things better

**Reply by egzumer** [reply]
Is this ramble addressed to me? :)

**Comment by Hunterfnet** [comment]
Hey [sfotis](https://github.com/sfotis) , i tested a bit and it's not bad a all , is good for start testing with others devices and firmwares with SSB/DSB band , will test more on free time , for now just the main contact and all seems on good route.
Ps: I need change the SSB mode on a frequency don't saved because if i do on a saved frequency then modifies all the saved channel and need reprogramm the channel , thats another heredated bug for solve that was hirritating me continuously . Just yesterday i changed definitely my point of view and started using your firmware mod for make little tests every few time with the IJV on DSB on my other walkie and more . Many Many thanks .   
Firmware-link  https://github.com/sfotis/uv-k5-firmware-ssbtx/releases
